New How-to template (#265)
thegooddocsproject/templates: how-to/how-to-template.md

> <!-- Copy this Template. -->

Reviewer: Suggestion (non-blocking): We probably don't need "copy this template" advice, as it is likely obvious to the reader. (We should aim to be concise.)

> <!-- Copy this Template. -->

Reviewer: Suggestion (non-blocking): We should be consistent with other templates regarding our standard advice.
Suggest looking at other published templates, and adopting the same conventions.

> <!-- Copy this Template. -->

Reviewer: Suggestion (blocking): We have previously decided against using HTML comments tags, and instead insert comments inside {curly brackets}.
Reason: We want the reader to be able to see these comments in whatever variant of the document they are reading. The HTML comments only visible if looking at the raw markdown, and are not visible when viewing in rendered HTML format.
This comment is applicable for all other instances of <! -- --> in this pull request.

> <!-- Describe the title of your article by replacing "How To Template" with the page name you want to publish to. -->

Reviewer: ```suggestion
```
Suggestion (non-blocking): Suggest that we don't need this advice, especially if we use a convention of putting variables inside curly brackets.

> # Title

Reviewer: ```suggestion
# {Title}
```
Suggestion (non-blocking): Suggest that all variables be put inside curly brackets. This is something we should confirm with our style committee.

> ::information_source: Before using this template, read the accompanying [how-to template guide](how-to-template-guide.md)

Reviewer 2: **nitpick:** A colon (:) appears at the beginning of the sentence in the output.

Author: Fixed

> ::information_source: Before using this template, read the accompanying [how-to template guide](how-to-template-guide.md)

Reviewer: Suggestion (non-blocking): We should be consistent with other templates regarding our standard format and advice. For instance, https://raw.githubusercontent.com/thegooddocsproject/templates/dev/quickstarts/quickstart-template.md has this sentence right at the top.

Suggest going through the other published templates and looking to make sure we are aligned.

## Overview

This guide explains how to {insert a brief description of the task}.

:information_source: Here are some examples:

- This guide explains how to comment on a pull request.

> - This guide explains how to create or delete a branch on Github.

Reviewer 3: I believe GitHub should be with a capital H.

Author: Fixed

- This guide explains how to create a repository.

> (Optional) Specify when and why your user might want to perform the task.

Reviewer: ```suggestion
{Optional: Specify when and why your user might want to perform the task.}
```
Suggestion: For consistency, we probably should insert instructions inside curly brackets. This comment applies to other Optional instructions in this template.

> ## Before you begin

Reviewer: ```suggestion
## {Optional} Before you begin
```
suggestion: Add an optional tag here.

<!-- Delete this section if your readers can go to the steps without requiring any prerequisite knowledge. -->

> Before you {insert a brief description of the task}, complete the following prerequisites:

Reviewer: suggestion (non-blocking):
```suggestion
Before you {insert a brief description of the task}, ensure:
```

- Prerequisite 1
- Prerequisite 2
- Prerequisite 3

## {Task name}

> 

Reviewer: ```suggestion

:information_source: If there is only one task in this guide, you can rename this section to "Instructions".

```
suggestion: Suggest adding this approach

1. {Write the step here. Use a verb to start.}

:information_source: You can use this format to describe your step:

Explanatory text

(Optional) Code sample or screenshot that helps your users complete this step.

> (Optional) The result of completing this step.

Reviewer: suggestion: An example will likely be very valuable here (or in the guide). Something like:
* Notice {button is visible at the bottom left of the display}
* Press {button}
* Observe {what has changed}
* Confirm {something has happened}

2.  {Write the step here. Use a verb to start.}

    2.1 {Substep 1}

    2.2 {Substep 2}

> 

Reviewer: ```suggestion

## {Task name 2}

```
suggestion: If we are suggesting that multiple tasks should be included, then we should have another task name heading.

Note: We probably should additionally be numbering this task heading.

> ## See also

Reviewer 4: suggestion, non-blocking: 'See also' and 'Additional resources' could be written exactly the same - DITA uses Related topics (or links) - maybe this is a question for the style guide.
